morpheus/stages/output/write_to_deltalake_stage.py

```diff
@@ -0,0 +1,209 @@
+                spark_dtype = StructField(col, StringType())
+            spark_schema.append(spark_dtype)
+        return StructType(spark_schema)
+
+    def _configure_databricks_connect(self,
+                                      databricks_host,
+                                      databricks_token,
+                                      databricks_cluster_id,
+                                      databricks_port,
+                                      databricks_org_id):
+        if (os.environ.get('DATABRICKS_HOST', None) is None and databricks_host is None):
+            raise Exception("Parameter for databricks host not provided")
+        if (os.environ.get('DATABRICKS_TOKEN', None) is None and databricks_token is None):
+            raise Exception("Parameter for databricks token not provided")
+        if (os.environ.get('DATABRICKS_CLUSTER_ID', None) is None and databricks_cluster_id is None):
+            raise Exception("Parameter for databricks cluster not provided")
+        host = None
+        cluster = None
+        token = None
+        config_file = "/root/.databricks-connect"
+        should_add = False
+        if (os.environ.get('DATABRICKS_HOST', None) is not None):
+            host = os.environ.get('DATABRICKS_HOST')
+        else:
+            host = databricks_host
+        if (os.environ.get('DATABRICKS_TOKEN', None) is not None):
+            token = os.environ.get('DATABRICKS_TOKEN')
+        else:
+            token = databricks_token
+        if (os.environ.get('DATABRICKS_CLUSTER_ID', None) is not None):
+            cluster = os.environ.get('DATABRICKS_CLUSTER_ID')
+        else:
+            cluster = databricks_cluster_id
+        config = """{
+                      "host": "@host",
+                      "token": "@token",
+                      "cluster_id": "@cluster_id",
+                      "org_id": "@org_id",
+                      "port": "@port"
+                }"""
+        config = config.replace("@host", host).replace("@token", token).replace("@cluster_id", cluster).replace(
+            "@org_id", databricks_org_id).replace("@port", databricks_port)
+
+        # check if the config file for databricks connect already exists
+        config_exist = os.path.exists(config_file)
+        if config_exist:
+            # check if the config being added already exists, if so do nothing
+            with open(config_file) as f:
+                if config in f.read():
+                    logger.info("Configuration for databricks-connect already exists, nothing added!")
+                else:
+                    logger.info("Configuration not found for databricks-connect, adding provided configs!")
+                    should_add = True
+        else:
+            should_add = True
+        if should_add:
+            with open(config_file, "w+") as f:
+                f.write(config)
+            logger.info("Databricks-connect successfully configured!")
```

Comment:
This function is repeated now in multiple places. It should be pulled out of the stages into a common location.